ladders/Controllers/ProfileController.cs

```diff
@@ -0,0 +1,182 @@
+
+            profileModel.UserId = GetMyName();
+            profileModel.Name = User.Claims.FirstOrDefault(c => c.Type == "name")?.Value;
+            profileModel.CurrentLadder = -1;
+            profileModel.Suspended = false;
+            return View(profileModel);
+        }
+
+        // POST: Profile/Create
+        // To protect from overposting attacks, please enable the specific properties you want to bind to, for 
+        // more details see http://go.microsoft.com/fwlink/?LinkId=317598.
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Create([Bind("Id,UserId,Name,Suspended,Availability,PreferredLocation")]
+            ProfileModel profileModel)
+        {
+            if (!ModelState.IsValid) return View(profileModel);
+
+            _context.Add(profileModel);
+            await _context.SaveChangesAsync();
+            return RedirectToAction(nameof(Index));
+        }
+
+        // GET: Profile/Edit/5
+        public async Task<IActionResult> Edit(int? id)
+        {
+            if (id == null) return NotFound();
+
+            var profileModel = await _context.ProfileModel.FindAsync(id);
+            if (profileModel == null) return NotFound();
+
+            if (!AmIAdmin() && !profileModel.UserId.Equals(GetMyName())) return NotFound();
+
+            return View(profileModel);
+        }
+
+        // POST: Profile/Edit/5
+        // To protect from overposting attacks, please enable the specific properties you want to bind to, for 
+        // more details see http://go.microsoft.com/fwlink/?LinkId=317598.
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Edit(int id,
+            [Bind("Id,Name,UserId,Availability,PreferredLocation,Suspended,CurrentLadder")]
+            ProfileModel profileModel)
+        {
+            if (id != profileModel.Id) return NotFound();
+
+            if (!ModelState.IsValid) return View(profileModel);
+
+            if (!AmIAdmin() && !profileModel.UserId.Equals(GetMyName())) return NotFound();
+
+            try
+            {
+                _context.Update(profileModel);
+                await _context.SaveChangesAsync();
+            }
+            catch (DbUpdateConcurrencyException)
+            {
+                if (!ProfileModelExists(profileModel.Id)) return NotFound();
+
+                throw;
+            }
+
+            return RedirectToAction(nameof(Index));
+        }
+
+        // GET: Profile/Delete/5
+        public async Task<IActionResult> Delete(int? id)
+        {
+            if (!AmIAdmin()) return RedirectToAction(nameof(Index));
+
+            if (id == null) return NotFound();
+
+            var profileModel = await _context.ProfileModel
+                .FirstOrDefaultAsync(m => m.Id == id);
+            if (profileModel == null) return NotFound();
+
+            return View(profileModel);
+        }
+
+        // POST: Profile/Delete/5
```

Comment:
Ditto on POSTing a DELETE

Reply:
These are how they are generated, not sure if we want to go away from their convention

Reply:
Scaffolding generates them as POST because browsers cant natively do DELETE from a simple form submission.  All the .NET apps are like this.